app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt

```diff
@@ -89,46 +90,34 @@ class SendPresenter(
         if (newValue != null) view?.showSourceToken(newValue)
     }
 
-    private var solToken: Token.Active? = null
-
-    private var inputAmount: String = "0"
-
-    private var tokenAmount: BigDecimal = BigDecimal.ZERO
-    private var usdAmount: BigDecimal = BigDecimal.ZERO
-
-    private var minRentExemption: BigInteger = BigInteger.ZERO
-
-    private var mode: CurrencyMode = CurrencyMode.Token(initialToken?.tokenSymbol ?: SOL_SYMBOL)
-
-    private var networkType: NetworkType = NetworkType.SOLANA
-
-    private var searchResult: SearchResult? = null
-
-    private var sendFee: SendFee? = null
+    private var state = SendState()
 
     private var calculationJob: Job? = null
     private var checkAddressJob: Job? = null
-
     private var feeJob: Job? = null
     private var feePayerJob: Job? = null
 
+    override fun setInitialToken(initialToken: Token.Active) {
+        state.updateInitialToken(initialToken)
+    }
+
     override fun loadInitialData() {
         launch {
             try {
                 view?.showFullScreenLoading(true)
-                view?.showNetworkSelectionView(initialToken.isRenBTC)
 
-                val userTokens = userInteractor.getUserTokens()
+                /*
+                 * We should find SOL anyway because SOL is needed for Selection Mechanism
+                 * */
                 val userPublicKey = tokenKeyProvider.publicKey
-                solToken = userTokens.find { it.isSOL && it.publicKey == userPublicKey }
-                    ?: throw IllegalStateException("No SOL account found")
-                token = initialToken ?: solToken
+                state.solToken = userInteractor.getUserTokens().find { it.isSOL && it.publicKey == userPublicKey }
+                token = state.initialToken ?: state.solToken
 
-                calculateTotal(sendFee = null)
-                view?.showNetworkDestination(networkType)
+                sendInteractor.initialize(state.solToken!!)
```

Comment:
i can suggest extracting `userInteractor.getUserTokens().find { it.isSOL && it.publicKey == userPublicKey }` to a local val and using it in both places to remove !!